x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/integrity/TransportRepositoryVerifyIntegrityCoordinationAction.java

```diff
@@ -0,0 +1,182 @@
+ * 2.0.
+ */
+
+package org.elasticsearch.repositories.blobstore.testkit.integrity;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionListenerResponseHandler;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.TransportAction;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.discovery.MasterNotDiscoveredException;
+import org.elasticsearch.injection.guice.Inject;
+import org.elasticsearch.repositories.RepositoriesService;
+import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskId;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportRequestOptions;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+public class TransportRepositoryVerifyIntegrityCoordinationAction extends TransportAction<
+    TransportRepositoryVerifyIntegrityCoordinationAction.Request,
+    RepositoryVerifyIntegrityResponse> {
+
+    /*
+     * Message flow: the coordinating node (the one running this action) forwards the request on to a master node which actually runs the
+     * verification. The master node in turn sends requests back to this node containing chunks of response, either information about the
+     * snapshots processed, or about the restorability of the indices in the repository, or details of any verification anomalies found.
+     * When the process is complete the master responds to the original transport request with the final results:
+     *
+     * +---------+                         +-------------+                           +--------+
+     * | Client  |                         | Coordinator |                           | Master |
+     * +---------+                         +-------------+                           +--------+
+     *      |                                     |                                       |
+     *      |-[REST request]--------------------->|                                       |
+     *      |                                     |---[master node request]-------------->| ----------------------\
+     *      |                                     |                                       |-| Initialize verifier |
+     *      |                                     |                                       | |---------------------|
+     *      |                                     |<--[SNAPSHOT_INFO chunk request]-------|
+     *      |<---[headers & initial JSON body]----|                                       |
+     *      |                                     |---[SNAPSHOT_INFO chunk response]----->| ------------------\
```

Comment:
Should these be `START_RESPONSE request` and `START_RESPONSE response`?

Reply:
Ah oops yes they should 2435c41d91965ec5e4fe8742a64f2815cb0af327